pandas/core/indexes/interval.py

```diff
@@ -691,61 +317,67 @@ def left(self):
         Return the left endpoints of each Interval in the IntervalIndex as
         an Index
         """
-        return self._left
+        return self._data._left
 
     @property
     def right(self):
         """
         Return the right endpoints of each Interval in the IntervalIndex as
         an Index
         """
-        return self._right
+        return self._data._right
 
     @property
     def closed(self):
         """
         Whether the intervals are closed on the left-side, right-side, both or
         neither
         """
-        return self._closed
+        return self._data._closed
 
     @property
     def length(self):
         """
         Return an Index with entries denoting the length of each Interval in
         the IntervalIndex
         """
-        try:
-            return self.right - self.left
-        except TypeError:
-            # length not defined for some types, e.g. string
-            msg = ('IntervalIndex contains Intervals without defined length, '
-                   'e.g. Intervals with string endpoints')
-            raise TypeError(msg)
+        return self._data.length
 
     @property
     def size(self):
-        # Avoid materializing self.values
-        return self.left.size
+        # Avoid materializing ndarray[Interval]
+        return self._data.size
 
     @property
     def shape(self):
-        # Avoid materializing self.values
-        return self.left.shape
+        # Avoid materializing ndarray[Interval]
+        return self._data.shape
+
+    @property
```

Comment:
> its a very odd thing to have generally for non-scalar based ndarray like things

A bit strange, yeah, since there are two representations: left & right arrays, or an array of Interval objects.

Anyway, on master this inherited from IndexOpsMixin, which reports the `ndarray[Interval]` version.
This is inconsistent with `Categorical`, which reports the itemsize of the codes. I'd also prefer that it's consistent with `nbytes`, which (should) report the memory used by the ExtensionArray, not the array of objects.